slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/ValidatorsDao.java

```diff
@@ -22,6 +22,21 @@
 
 public class ValidatorsDao {
 
+  public Validator insertIfNotExist(final Handle handle, final Bytes validator) {
+    final List<Validator> result =
+        handle
+            .createUpdate("INSERT INTO validators (public_key) VALUES (?) ON CONFLICT DO NOTHING")
+            .bind(0, validator)
+            .executeAndReturnGeneratedKeys()
+            .mapToBean(Validator.class)
+            .list();
+
+    if (result.isEmpty()) {
+      return retrieveValidators(handle, List.of(validator)).get(0);
```

Comment:
This might get a bit expensive retrieving the entire set of validators over and over if the validators already exist.